src/plugins/dam/sagas/uploadFileFlow.js

```diff
@@ -47,32 +47,18 @@ export default function* (hashName, fileInfo, fileBuffer, preSignedUrl, variant)
         },
       });
 
+      if (response.status !== 200) {
+        throw new Error(`dam upload file flow failed on try #${retries}. trying again.`);
+      }
+
       const { version } = variant;
       const { asset } = fileInfo;
-      const previewUrl = damUrl(asset, version, 'sm');
       const { type } = asset.get('_id');
 
-      try {
-        if (type === 'image') {
-          yield checkImage(previewUrl);
-        } else {
-          const assetResponse = yield call(fetch, previewUrl, { method: 'get' });
-          if (assetResponse.status !== 200) {
-            throw new Error('dam upload file flow failed first try. trying again.');
-          }
-        }
-      } catch (e) {
-        // Make one more try after 2 secs because sometimes amazon
-        // lags on making the asset available... wtf amazonians
-        yield delay(2000);
```

Comment:
@efjacobson this delay was kind've important because sometimes it took a moment for an image preview to become available. without it, we could go over the MAX_RETRIES and possibly alert a user that the image failed even though it was possibly successful.

Reply:
@rsumilang like, if checkImage rejects? or do you mean the presigned PUT. it does still have the delay at the beginning of the loop if it is not the first retry

Reply:
@efjacobson the `put` was usually good but checking if an image was available for previewing via `checkImage` had a high fail rate on the initial check as it took amazon a while to make asset available via url. I'd feel more comfortable with this change if we had a higher `MAX_RETRIES` for images.

Reply:
@rsumilang ok yeah i can whip something up. think i'll only retry the PUT if it fails though, seems odd to do that again if it was just the checkImage that failed

Reply:
@rsumilang lmk what you think now. the delay is now 2 seconds per retry, max of 5, and i wrapped the `yield put(e)` in its own try since it is `put` is only supposed to be used for plain actions and was choking on exceptions